File: backend/src/main/java/edu/cit/lim/gymtrack/repository/MembershipRepository.java
package edu.cit.lim.gymtrack.repository;

import edu.cit.lim.gymtrack.entity.Membership;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;
import java.util.Optional;

public interface MembershipRepository extends JpaRepository<Membership, Long> {
    Optional<Membership> findTopByUserIdOrderByEndDateDesc(Long userId);
    List<Membership> findByUserIdOrderByEndDateDesc(Long userId);
}

File: mobile/app/src/main/java/edu/cit/lim/gymtrack/mobile/feature/membership/MembershipRepository.kt
package edu.cit.lim.gymtrack.mobile.feature.membership

import edu.cit.lim.gymtrack.mobile.data.model.MembershipResponse
import edu.cit.lim.gymtrack.mobile.data.remote.ApiResponses
import edu.cit.lim.gymtrack.mobile.data.remote.ApiService

class MembershipRepository(private val apiService: ApiService) {

    suspend fun myMembership(): MembershipResponse =
        ApiResponses.unwrap(apiService.myMembership())
}
